internal/kv/mem/mem_kv.go

```diff
@@ -145,7 +145,26 @@ func (kv *MemoryKV) MultiRemoveWithPrefix(keys []string) error {
 	panic("not implement")
 }
 func (kv *MemoryKV) MultiSaveAndRemoveWithPrefix(saves map[string]string, removals []string) error {
-	panic("not implement")
+	kv.Lock()
+	defer kv.Unlock()
+
+	for key, value := range saves {
```

Comment:
why are we save first then remove?
In current use case maybe save and remove are not under same prefix, but what if user of this function use the same prefix?

Reply:
Yes. I will change it.